### ISSUE TITLE
[blogger] Google Drive image uploads not being automatically detected for mobile or social media thumbnails

### ISSUE THREAD

**Issue by nvella** [OpenLiveWriter/OpenLiveWriter#814]
Images uploaded in Blogger posts are currently not being detected for thumbnails on mobile or on external social media sites.

To reproduce:
- Post an image to a Blogger blog.
- Navigate to `BLOG_URL/?m=1` to view the mobile view
- Thumbnails do not appear
- Attempting to link the post on social media sites like Facebook results in no image preview being generated.

I've inspected the HTML source of the posts on Blogger and it seems it is successfully including the Open Graph tag for the image preview. Facebook may not be allowed to generate previews for googleusercontent, this could be one reason why it fails.

**Comment by wcbx** [comment]
I think it would be best to use the Google Photos API rather than Google Drive. I personally will not be using Open Live Writer for blogger until that gets implicated. I appreciate your work allowing uploading of images to Google Drive but I believe that's just the wrong direction because Blogger is built to use Google Photos.

**Comment by PeterMacej** [comment]
@wcbx
> 
> 
> that's just the wrong direction because Blogger is built to use Google Photos.

That's not true. This is not related to this issue, but for clarification. Blogger doesn't use Google Photos natively. Just create a new blogger post via the web interface and insert an image in it. No need to publish, just save it as a draft. You will not find the image in Google Photos. Blogger stores images in its own storage place that has no special name, as far as I know. And the only place where you can see such blogger images is [Google Album Archive](https://get.google.com/albumarchive). See https://support.google.com/blogger/answer/41641?hl=en

This mechanism was not possible from Open Live Writer (or any other blogger client), so it stored the photos to Google Photos. This was possible until March 2019. After that, it's no longer possible to upload a photo to Google Photos programmatically and get its permanent URL usable in a blog post. That's what @nvella tried as the first option. It seems that this is not possible unless Google changes the API (not very likely).

So Google Drive came as the next reliable solution.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
